added actions and filters to modify tax calculation for shipping taxes and fee taxes #16938

stiletagegmbh opened the pull request:
Filters and actions are needed to ensure correct tax calculation by
German law.

The filter in class-wc-cart-totals.php is used to modify fee tax
calculation during checkout and in cart.

The actions in the other files are used to modify tax calculation in
backend for manual orders.

Commit:
added actions and filters to modify tax calculation for shipping taxes and fee taxes #16938

Filters and actions are needed to ensure correct tax calculation by
German law.

The filter in class-wc-cart-totals.php is used to modify fee tax
calculation during checkout and in cart.

The actions in the other files are used to modify tax calculation in
backend for manual orders.

mikejolley commented on includes/class-wc-cart-totals.php:
```diff
@@ -313,6 +313,8 @@ protected function get_fees_from_cart() {
 				}
 			}
 
+			$fee->taxes = apply_filters( 'woocommerce_get_fees_from_cart_taxes', $fee->taxes, $fee, $this );
```
I'd rename this for clarity since the method name is not unique.

`woocommerce_cart_totals_get_fees_from_cart_taxes`

The other actions and filters don't have this prefix; they existed before cart-totals class was added.

stiletagegmbh replied:
Of course, I agree.
You can rename it to:
woocommerce_cart_totals_get_fees_from_cart_taxes

Commit:
#16938 rename filter for clarity

stiletagegmbh commented:
I've added a commit, I renamed the filter name for clarity as you suggested here: https://github.com/woocommerce/woocommerce/pull/16969/files/73bc8b8e6c25e05c4a04451b932592d0df0fb43d#r142166829
